src/lib.rs

```diff
@@ -460,41 +478,48 @@ struct CLruNode<K, V> {
     value: V,
 }
 
-/// An LRU cache with constant time operations.
+/// A weighted LRU cache with constant time operations.
```

Comment:
Are they still constant? `put_with_weight()` has to potentially "make room" for an entry when the cache is near full capacity.

This can end-up being `O(capacity)` in the worst case. That is, when adding a big entry to a near-full cache, which requires vacating all of the elements, and all those elements have unitary / small weight.

Better mentioning that here.

Reply:
So, the operations that were previously constant, remains constant, but the newly introduced `put_with_weight` is effectively, not constant

Reply:
I have reworked the doc